tools/releasing/update_branch_version.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+##
+## Variables with defaults (if not overwritten by environment)
+##
+MVN=${MVN:-mvn}
+
+##
+## Required variables
+##
+OLD_VERSION=${OLD_VERSION}
+NEW_VERSION=${NEW_VERSION}
+
+if [ -z "${OLD_VERSION}" ]; then
+    echo "NEW_VERSION was not set."
+    exit 1
+fi
+
+if [ -z "${NEW_VERSION}" ]; then
+    echo "NEW_VERSION was not set."
+    exit 1
+fi
+
+# fail immediately
+set -o errexit
+set -o nounset
+
+CURR_DIR=`pwd`
+BASE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+PROJECT_ROOT="${BASE_DIR}/../../"
+
+# Sanity check to ensure that resolved paths are valid; a LICENSE file should aways exist in project root
+if [ ! -f ${PROJECT_ROOT}/LICENSE ]; then
+    echo "Project root path ${PROJECT_ROOT} is not valid; script may be in the wrong directory."
+    exit 1
+fi
+
+###########################
+
+cd ${PROJECT_ROOT}
+
+# change version in all pom files
+mvn versions:set -DgenerateBackupPoms=false -DnewVersion=${NEW_VERSION}
+
+# change version in Python SDK's setup.py file
+perl -pi -e "s#version=\'$OLD_VERSION\'#version=\'$NEW_VERSION\'#" flink-ml-python/setup.py
```

Comment:
We might need also modify this line to 
```
SHORT_OLD_VERSION=${OLD_VERSION%%"-SNAPSHOT"}
SHORT_NEW_VERSION=${NEW_VERSION%%"-SNAPSHOT"}
perl -pi -e "s#version=\'$SHORT_OLD_VERSION\'#version=\'$SHORT_NEW_VERSION\'#" flink-ml-python/pyflink/ml/version.py
```

Reply:
After going through the create_python_sdk_release.sh used by flink-statefun, I realize that the development of this script depends a lot on the directory structure of flink-ml-python. Since I am not the developer of flink-ml-python and I have not experience with python release, it is probably better for someone else to create this script for Flink ML.

I have removed this script from this PR.

Reply:
I have updated this line to the following lines, similar to the approach used by `flink/tools/releasing/update_branch_version.sh`.

```
perl -pi -e "s#^__version__ = \".*\"#__version__ = \"${NEW_VERSION}\"#" flink-ml-python/pyflink/ml/version.py
perl -pi -e "s#-SNAPSHOT#\\.dev0#" flink-ml-python/pyflink/ml/version.py
```